src/protobuf-net.Reflection/CSharpCodeGenerator.cs

```diff
@@ -608,7 +610,7 @@ protected override void WriteField(GeneratorContext ctx, FieldDescriptorProto fi
             }
             else
             {
-                tw = ctx.Write($"{GetAccess(GetAccess(field))} {typeName} {Escape(name)} {{ get; set; }}");
+                tw = ctx.Write($"{GetAccess(GetAccess(field))} {(typeName + (IsNullableType(ctx, field) ? "?":""))} {Escape(name)} {{ get; set; }}");
```

Comment:
can this be a separate `{}` section, to avoid the string concat? not really an important cost in the context, but:

``` c#
{typeName}{(IsNullableType(ctx, field) ? "?" : "")}
```